include/turtle/stream.hpp

```diff
@@ -106,7 +106,13 @@ namespace detail {
     {
         s << '"' << str << '"';
     }
-    inline void serialize(stream& s, const char* const str) { s << '"' << str << '"'; }
+    inline void serialize(stream& s, const char* const str)
+    {
+        if (nullptr != str)
```

Comment:
What about
```suggestion
        if (str)
```
?
Or is that considered back practice these days (sorry, I haven’t done any C++ in years)?

I think it should at least be
```suggestion
        if (str != nullptr)
```
?

Reply:
Again, merely force of habbit. I'm used to the defensive style of putting the constant on the left in comparisons, so that when one accidentally types `if (nullptr = str)` instead of `if (nullptr == str)` the compiler catches it, unlike `if (str = nullptr)` which silently does something unintended. Again, if treating pointers as if they are booleans is your preferred style then I'm very happy to change this to match.